numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4166,6 +4166,54 @@ small_correlate(const char * d_, npy_intp dstride,
     }
 }
 
+/*
+*/
+
+/* A clone function for the datetime dtype c_metadata */
+static NpyAuxData *
+_datetime_dtype_metadata_clone(NpyAuxData *data)
+{
+    PyArray_DatetimeDTypeMetaData *newdata =
+        (PyArray_DatetimeDTypeMetaData *)PyArray_malloc(
+                        sizeof(*newdata));
+    if (newdata == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+
+    memcpy(newdata, data, sizeof(*newdata));
+
+    return (NpyAuxData *)newdata;
+}
+
+/*
+ * Initializes the c_metadata field
+ */
+static NpyAuxData*
+_get_builtin_datetime_metadata(void)
```

Comment:
Optional: could change the signature to `_create_datetime_metadata(base, num)`, and move the actual defaults to the call site

Reply:
adopted